spec/support/html_matchers.rb

```diff
@@ -6,8 +6,11 @@
 RSpec::Matchers.define :have_valid_html do
   match do |actual|
     @validator = W3CValidators::NuValidator.new
-    vcr_options = { match_requests_on: %i[uri method html_body_content], record: :new_episodes }
-    @results = VCR.use_cassette("w3c_valid_content", vcr_options) do
+    vcr_options = { match_requests_on: %i[html_body_content], record: :once}
+
+    cassette_name = method_missing(:class).metadata[:rerun_file_path].split("/").last.gsub(".rb", "")
```

Comment:
Might be useful to have a comment with the expected result as there's quite a lot happening on this line.